accepted/00000-remote-command-auditing.md

```diff
@@ -0,0 +1,106 @@
+- A `Remote Command` tab is present in the details section of a traditional or minion system (or a `System Set Manager` group)
+- The API exposes the `system.scheduleScriptRun` to issue a command
+- (Only for minions) a `Salt > Remote Commands` page to launch a command on minions by leveraging [Salt targeting](https://docs.saltstack.com/en/latest/topics/targeting/index.html) to identify the target(s)
+
+Each user can only target the systems within the same organization that the systems are registered to, provided that the user has a particular role:
+- Every user that has `SUSE Manager Administrator` or `Organization Administrator` role can access the tab and run a remote command on the selected system or a group of systems
+- The API is always exposed but checks for the role of the user supplying the command after the API is called
+- For every user registered to SUSE Manager, independent of the role, the menu item `Salt > Remote Commands` is accessible. Only users with the `SUSE Manager Administrator` or `Organization Administrator` role, though, can impact systems using this feature, whereas all other users cannot target any system (`"no systems found"`).
+
+Every system outside the organization of the user issuing the command is not impacted nor is it visible by the user.
+
+The motivations behind the changes introduced in this RFC are _security_ and _auditing_. Any command issues via the `Remote Command` will be run with `root` privileges on the target system(s). Additionally, the user that issued the command is not logged along with the command issues and its output.
+Many customers claim to require this feature before migrating from traditional clients to Salt minions.
+
+Reference: https://fate.suse.com/325624 https://fate.suse.com/327158 
+
+In order to overcome the above limitations, in this RFC we are going to introduce:
+
+- Role-based restriction for `Remote Command`: a new role will be introduced. Every user associated with this role (in addition to either `SUSE Manager Administrator` or `Organization Administrator`) can access the `Remote Command` feature and issue commands. Every other user will not see the `Remote Command` tab nor the `Salt > Remote Commands` menu item.
+- Auditing for `Remote Command` and `Salt > Remote Commands`: the product will log the user that has issued the command, the command itself and its result.
+
+# Detailed design
+[design]: #detailed-design
+
+The first step is to introduce a new role in the roles list: `Remote Command Administrator` (`RhnUserGroupType` table).
+The role is a normal role (comparable to `Config Administrator`, `Image Administrator`, etc) and not an `Administrative Role` (`SUSE Manager Administrator` or `Organization Administrator`). If a user has a `Remote Command Administrator` role but not one of the `Administrative Role`s, then the user cannot see any system and thus should behave like the other normal roles.
```

Comment:
Keep in mind that even if a user has no administrative role, it might be assigned to a "System Group". Then the user is allow to manage that particular group according to the normal roles the user has.

So, I would say we need to refactor this because it's not true that "... If a user has a `Remote Command Administrator` role but not one of the `Administrative Role`s, then the user cannot see any system ...".

I also agree with @mcalmer that maybe it's enough to use the existing "Configuration Administration" which currently allows the user to manage the "Configuration States", since running Salt states is basically the same (in the context of permissions and accessibility) than running an execution module of Salt.

Why someone could think on allowing "Remote Execution Commands" and not "Configuration States" for a system? Both roles do allow to basically run whatever command as "root" on the targeted minion.

Reply:
You're right, adding it to spec.